js/components/input.jsx

```diff
@@ -23,23 +23,34 @@ var Input = React.createClass({
   },
 
   getInitialState: function() {
-    return {message: ''};
+    return {
+      message: '',
+      commandCharacter: SettingsStore.commandCharacter()
+    };
+  },
+
+  componentDidMount: function() {
+    SettingsStore.on('change', this.settingsStoreChange);
   },
 
   _onChange: function(event) {
-    this.setState({message: event.target.value});
+    this.setState({message: event.target.value, commandCharacter: this.state.commandCharacter});
   },
 
   _onKeyDown: function(event) {
     if (event.keyCode === KeyCodes.enter) {
       event.preventDefault();
-      if (this.state.message[0] === CommandCharacter) {
+      if (this.state.message[0] === this.state.commandCharacter) {
```

Comment:
So from what I understand it could be but then the component one level up would have to listen to the settings store and trigger and update. If/when we add more settings that are using in other components it may well make sense to listen to the SettingsStore at a higher level, but doesn't really right now.